query.go

```diff
@@ -98,12 +107,92 @@ func (u *UtxorpcClient) GetUtxoByRef(
 		Index: txIndex,
 	}
 	req := &query.ReadUtxosRequest{Keys: []*query.TxoRef{txoRef}}
-	return u.ReadUtxos(req)
+	return u.ReadUtxosWithContext(ctx, req)
+}
+
+type TxoReference struct {
+	TxHash string
+	Index  uint32
+}
+
+func (u *UtxorpcClient) GetUtxosByRefs(
+	refs []TxoReference,
+	batchSize *int,
+) (*connect.Response[query.ReadUtxosResponse], error) {
+	return u.GetUtxosByRefsWithContext(context.Background(), refs, batchSize)
+}
+
+func (u *UtxorpcClient) GetUtxosByRefsWithContext(
+	ctx context.Context,
+	refs []TxoReference,
+	batchSize *int,
+) (*connect.Response[query.ReadUtxosResponse], error) {
+	if len(refs) == 0 {
+		return nil, errors.New("no transaction references provided")
+	}
+
+	defaultBatchSize := 100
+	if batchSize != nil && *batchSize > 0 {
+		defaultBatchSize = *batchSize
+	}
+
+	allTxoRefs := make([]*query.TxoRef, 0, len(refs))
+	for _, ref := range refs {
+		var txHashBytes []byte
+		txHashBytes, err := hex.DecodeString(ref.TxHash)
+		if err != nil {
+			return nil, err
+		}
+		txoRef := &query.TxoRef{
+			Hash:  txHashBytes,
+			Index: ref.Index,
+		}
+		allTxoRefs = append(allTxoRefs, txoRef)
+	}
+
+	if len(allTxoRefs) <= defaultBatchSize {
+		req := &query.ReadUtxosRequest{Keys: allTxoRefs}
+		return u.ReadUtxosWithContext(ctx, req)
+	}
+
+	var allResults []*query.AnyUtxoData
+	for i := 0; i < len(allTxoRefs); i += defaultBatchSize {
```

Comment:
This batching should really be done by the consumer, preferably with some kind of caching. There's no batching or limits in the protocol, which means there's no backend pagination or batch management. UTxO RPC APIs will have to get all of the information and then post-filter it in the SDK client, without indicators of what's been delivered, the total number of items, current page in the batch, etc.